src/group/mls_group/new_from_welcome.rs

```diff
@@ -73,24 +73,27 @@ impl MlsGroup {
             .iter()
             .position(|e| e.extension_type() == ExtensionType::RatchetTree);
         let ratchet_tree_extension = if let Some(i) = ratchet_tree_ext_index {
-            let extension = group_info.extensions_mut().remove(i);
             // Throw an error if we there is another ratchet tree extension.
             // We have to see if this makes problems later as it's not something
             // required by the spec right now.
             if group_info
                 .extensions()
                 .iter()
-                .any(|e| e.extension_type() == ExtensionType::RatchetTree)
+                .filter(|e| e.extension_type() == ExtensionType::RatchetTree)
```

Comment:
We're iterating twice over the extensions here. I think you can optimize by filtering first, then checking for duplicates and then using the one that's left (if there is one).

Reply:
Where else are we iterating? I'm not really concerned about performance here, the list of extensions should be < 10 most of the time.